x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java

```diff
@@ -264,6 +264,43 @@ public void onFailure(Exception e) {
         }
     }
 
+    /**
+     * Sets the error for the {@link ConnectorSyncJob} in the underlying index.
+     * This also sets the {@link ConnectorSyncStatus} to 'ERROR'.
+     *
+     * @param connectorSyncJobId     The id of the connector sync job object.
+     * @param error                  The error to set.
+     * @param listener               The action listener to invoke on response/failure.
+     */
+    public void updateConnectorSyncJobError(String connectorSyncJobId, String error, ActionListener<UpdateResponse> listener) {
+        final UpdateRequest updateRequest = new UpdateRequest(CONNECTOR_SYNC_JOB_INDEX_NAME, connectorSyncJobId).setRefreshPolicy(
+            WriteRequest.RefreshPolicy.IMMEDIATE
+        )
+            .doc(
+                Map.of(
+                    ConnectorSyncJob.ERROR_FIELD.getPreferredName(),
+                    error,
```

Comment:
What happens if we want to reset the error to `null`? As a sideeffect we would set the sync job to error (not sure if it's possible with sync job, setting error to null for connector is possible)

Followup: let's discuss the allowed state transitions

Reply:
I'm actually not sure, if you should be able to "reset" a sync job. Currently, if a sync job fails it stops and cannot be picked up again as we don't do checkpointing (we've incremental syncs, but for a different use case).